src/amqp-channel.ts

```diff
@@ -274,12 +274,14 @@ export default class AMQPChannel {
    * @param [immediate] - if the message should be returned if it can't be delivered to a consumer immediately (not supported in RabbitMQ)
    * @return - fulfilled when the message is enqueue on the socket, or if publish confirm is enabled when the message is confirmed by the server
    */
-  basicPublish(exchange: string, routingKey: string, data: string|Uint8Array|ArrayBuffer|null, properties: AMQPProperties = {}, mandatory = false, immediate = false) {
+  basicPublish(exchange: string, routingKey: string,
+               data: string|Uint8Array|ArrayBuffer|null|Record<string|symbol|number, unknown>,
```

Comment:
i think it's a mistake to allow "objects", and that the lib autoamtically encodes it as json.